src/generator/pollers.ts

```diff
@@ -53,50 +56,60 @@ type ${poller.name} struct {
 	pt pollingTracker
 }
 
-// Done returns true if the polling operation has terminated either in a success case or failure case,
-// otherwise it will return false
-func (p *${poller.name}) Done() bool {
-	return p.pt.hasTerminated()
-}
-
-// ID ... 
-func (p *${poller.name}) ID() string {
-	return ""
+// Poll returns false if there was an error or polling has reached a terminal state
+func (p *${poller.name}) Poll(ctx context.Context) bool {
+	if done, err := p.done(ctx); err != nil {
+		return false
+	} else {
+		return !done
+	}
 }
 
-func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
-	if done, err := p.done(ctx); !done || err != nil {
-		return nil, err
-	}
+// Response returns the final response from the polling operation
```

Comment:
We might want to change this to return the last response for consumers that need to write custom end-to-end polling (kubernetes); this would mean that we'd never return a nil response envelope as it would always have a raw response, not sure how I feel about that.  We can discuss later and update/change as required in a future PR.

Reply:
I think my comment was misleading, I've updated it now.